File: java/src/main/java/com/agview/api/example/PremisePostHandler.java
package com.agview.api.example;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.util.Collection;
import java.util.HashMap;
import java.util.function.Function;
import java.util.stream.Collectors;

public class PremisePostHandler {

    private final HttpClient httpClient;
    private final Arguments connectionInfo;
    private final AccessTokenHandler accessTokenHandler;
    private final PremiseDbHandler premiseDbHandler;

    public PremisePostHandler(HttpClient httpClient,
                              Arguments connectionInfo,
                              AccessTokenHandler accessTokenHandler,
                              PremiseDbHandler premiseDbHandler) {
        this.httpClient = httpClient;
        this.connectionInfo = connectionInfo;
        this.accessTokenHandler = accessTokenHandler;
        this.premiseDbHandler = premiseDbHandler;
    }

    public Collection<CreatedPremise> createPremises() {
        try {
            Collection<Premise> premises = premiseDbHandler.getPremisesToLoad();
            Collection<PremiseAddress> premiseAddresses = premiseDbHandler.getPremiseAddressesToLoad();

            var premiseAddressByUsdaPin = premiseAddresses.stream().collect(Collectors.toMap(PremiseAddress::getUsdaPin, Function.identity()));

            var postRequestBody = premises.stream().map(premise ->
                    new HashMap<String, Object>() {
                        {
                            put("usdaPin", premise.getUsdaPin());
                            put("premName", premise.getPremName());
                            put("species", premise.getSpecies());
                            put("iceContactPhone", premise.getIceContactPhone());
                            put("iceContactEmail", premise.getIceContactEmail());
                            put("locationType", premise.getLocationType());
                            put("siteCapacityNumberBarns", premise.getSiteCapacityNumberBarns());
                            put("siteCapacityNumberAnimals", premise.getSiteCapacityNumberAnimals());
                            put("numberOfAnimalsOnSite", premise.getNumberOfAnimalsOnSite());
                            put("streetAddress", premiseAddressByUsdaPin.get(premise.getUsdaPin()).getStreetAddress());
                            put("city", premiseAddressByUsdaPin.get(premise.getUsdaPin()).getCity());
                            put("state", premiseAddressByUsdaPin.get(premise.getUsdaPin()).getState());
                            put("zip", premiseAddressByUsdaPin.get(premise.getUsdaPin()).getZip());
                        }
                    }
            )
                    .collect(Collectors.toList());

            var objectMapper = new ObjectMapper();
            var bodyJsonStr = objectMapper
                    .writerWithDefaultPrettyPrinter()
                    .writeValueAsString(postRequestBody);
            var accessToken = accessTokenHandler.getNewAccessToken();
            var request = HttpRequest.newBuilder()
                    .uri(new URI(connectionInfo.getBaseUrl() + "/api/v1/prems/"))
                    .headers("Content-Type", "application/json",
                            "Authorization", "Bearer "+accessToken.getAccess())
                    .POST(HttpRequest.BodyPublishers.ofString(bodyJsonStr))
                    .build();
            var future = httpClient.sendAsync(request, HttpResponse.BodyHandlers.ofString())
                            .thenApply(HttpResponse::body);
            var responseObjectMapper =
                    new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);

            return responseObjectMapper.readValue(future.get(), new TypeReference<>(){});
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

}

File: c-sharp/Api/PremisePostHandler.cs
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

namespace Npb.Agview.Api.Example
{
    public class PremisePostHandler
    {
        private readonly HttpClient _httpClient;
        private readonly AccessTokenHandler _accessTokenHandler;
        private readonly string _baseUrl;
        private readonly PremiseDbHandler _premiseDbHandler;

        public PremisePostHandler(HttpClient httpClient,
            AccessTokenHandler accessTokenHandler,
            string baseUrl,
            PremiseDbHandler premiseDbHandler)
        {
            _httpClient = httpClient;
            _accessTokenHandler = accessTokenHandler;
            _baseUrl = baseUrl;
            _premiseDbHandler = premiseDbHandler;
        }

        public async Task<List<CreatedPremise>> CreatePremises()
        {
            var premises = _premiseDbHandler.GetPremisesToLoad();
            var premiseAddresses = _premiseDbHandler.GetPremiseAddressesToLoad();
            var premiseAddressByUsdaPin = premiseAddresses.ToDictionary(k => k.UsdaPin, v => v);

            var requestBody = premises.Select(premise => new
            {
                usdaPin = premise.UsdaPin,
                premName = premise.PremName,
                species = premise.Species,
                iceContactPhone = premise.IceContactPhone,
                iceContactEmail = premise.IceContactEmail,
                locationType = premise.LocationType,
                siteCapacityNumberBarns = premise.SiteCapacityNumberBarns,
                siteCapacityNumberAnimals = premise.SiteCapacityNumberAnimals,
                numberOfAnimalsOnSite = premise.NumberOfAnimalsOnSite,
                streetAddress = premiseAddressByUsdaPin[premise.UsdaPin].StreetAddress,
                city = premiseAddressByUsdaPin[premise.UsdaPin].City,
                state = premiseAddressByUsdaPin[premise.UsdaPin].State,
                zip = premiseAddressByUsdaPin[premise.UsdaPin].Zip
            });

            var requestBodyStr = JArray.FromObject(requestBody).ToString();

            var accessToken = _accessTokenHandler.GetNewAccessToken().Result;
            _httpClient.DefaultRequestHeaders.Authorization =
                    new AuthenticationHeaderValue("Bearer", accessToken.Access);
            var response = await _httpClient.PostAsync($"{_baseUrl}/api/v1/prems/", new StringContent(requestBodyStr, Encoding.UTF8, "application/json"));

            var responseBodyStr = response.Content.ReadAsStringAsync().Result;

            return JsonConvert.DeserializeObject<List<CreatedPremise>>(responseBodyStr);
        }
    }
}
